remote_config/src/common.h

```diff
@@ -63,6 +66,25 @@ class FutureData {
   static FutureData* s_future_data_;
 };
 
+template <typename TConfigUpdate, typename TRemoteConfigError>
+class LambdaConfigUpdateListener {
+ public:
+  LambdaConfigUpdateListener(
+      std::function<void(TConfigUpdate, TRemoteConfigError)> callback)
```

Comment:
when referencing the function, I think this is ConfigUpdate&& to match the function's type in the public API. Same change for line 85. I'm unsure what this means for onUpdate, as I haven't used the && address reference feature much.

Reply:
I think the compiler is able to infer the type of lvalue from the std::function since the integration test runs and is successful. Also adding the double ampersand ends up breaking the build (see the attached photo) 
![image](https://user-images.githubusercontent.com/93349266/232948018-78da44c4-0491-43b6-be80-92d1c60177f0.png)

Reply:
Removed the listener wrapper